ipn/ipnlocal/local.go

```diff
@@ -4781,12 +4799,19 @@ func (b *LocalBackend) Logout(ctx context.Context) error {
 func (b *LocalBackend) setNetInfo(ni *tailcfg.NetInfo) {
 	b.mu.Lock()
 	cc := b.cc
+	refresh := b.refreshReportForExitNodeSuggestion
+	b.refreshReportForExitNodeSuggestion = false
 	b.mu.Unlock()
 
 	if cc == nil {
 		return
 	}
 	cc.SetNetInfo(ni)
+	if refresh && shouldAutoExitNode() {
+		b.mu.Lock()
+		b.suggestExitNodeLocked()
```

Comment:
no report inside ni

Reply:
Oh, I see, `tailcfg.NetInfo` is what the client sends up to control, like a digested version of the report. We could probably store that in `LocalBackend` and then don't even need to dig the last report out of `MagicConn` in `suggestExitNodeLocked`. Sadly, I don't see a way to get `ni` back out of `cc` for the other code paths.

That sounds like a bunch of busywork though and the current thing works, so that'd be better as a cleanup in another PR.